tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# pylint: disable=redefined-outer-name,unused-argument,duplicate-code
+
+"""Integration test relation file."""
+
+import logging
+
+import jenkinsapi.jenkins
+import requests
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+from pytest_operator.plugin import OpsTest
+
+import jenkins
+
+from .constants import ALL_PLUGINS
+from .helpers import install_plugins
+from .types_ import UnitWebClient
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def test_jenkins_deploy_with_plugins(ops_test: OpsTest, jenkins_image: str):
+    """
+    arrange: given a juju model.
+    act: deploy Jenkins, instantiate the Jenkins client and install the plugins.
+    assert: the deployment has no errors.
+    """
+    # Ignoring assignment lint error due to knowing there will always be a model.
+    model: Model = ops_test.model  # type: ignore[assignment]
```

Comment:
I believe there's a fixture for model

Reply:
there is, but as I need to use a totally diffferent model form that in the fixture, that deploys the local charm at first. this test requires a totally different module.

Reply:
How come? Isn't it empty?
```
@pytest.fixture(scope="module", name="model")
def model_fixture(ops_test: OpsTest) -> Model:
    """The testing model."""
    assert ops_test.model
    return ops_test.model
```